main.go

```diff
@@ -199,30 +168,33 @@ func main() {
 		})
 	}
 	{
-		var cancel func()
-
-		if opts.Duration != 0 {
-			ctx, cancel = context.WithTimeout(ctx, opts.Duration)
-		} else {
-			ctx, cancel = context.WithCancel(ctx)
-		}
-
-		t := time.NewTicker(opts.Period)
+		if opts.AddressRead != nil {
```

Comment:
If read address is not there, it doesn't schedule any actors.

Reply:
Yes this is the right change, with the previous changes, we were leaking when not reading. However, you have an `if` fully nestled within another block. I think the outside block should simply be the if, that was the alignment is easier to read as well